src/faction.cpp

```diff
@@ -26,7 +26,7 @@ std::string invent_adj();
 
 constexpr inline unsigned long mfb( const int v )
 {
-    return 1 << v;
+    return static_cast<unsigned long>( 1 << v );
```

Comment:
This is wrong. The cast was already implicit by the return. Pretty sure the static analyzer warned you about shifting the 1 which is an integer; because when v is bigger or equal to than the width of the integer you get UB. Youll want to just cast the 1 prior to shifting so the shifting will operate on a unsigned long of which the result is a unsigned long.

Reply:
What are you suggesting?

```CPP
    return 1ul << v ;
    return static_cast<unsigned long>( 1 )<< static_cast<unsigned long>( v );
```

Reply:
Initially `return static_cast<unsigned long>(1) << v;`. There should be no need to cast v.
However your `return 1ul << v;` should be functionally identical and looks a lot less verbose and thus prefered imo.